client/js/components/common/fill_the_blank/fill_the_blank_word_chain.jsx

```diff
@@ -1,38 +1,130 @@
 import React                            from 'react';
+import ReactDOM                         from 'react-dom';
 
 import { FillTheBlankDraggableWord }    from '../draggable_word';
 import { FillTheBlankWordDropZone }     from '../drop_zones';
 import Word                             from '../word';
+import { beginWrap, endWrap }           from '../movable_words/word_chain';
 
-export default (props) => {
-  // range is times two minus 1 so that we have the correct spaces for the dropzones.
-  return <div className="c-answers">
-    <div className="c-blank-answers">
-      {
-        _.map(props.sentenceWords, (wordHtml, index) => {
-          if(wordHtml.indexOf("interaction-placeholder") >= 0) {
-            if(!_.isEmpty(props.selectedAnswer)) {
-              return <div key={`${props.selectedAnswer.id}-${index}`} className="c-blank-drop-zone">
-                <FillTheBlankDraggableWord
-                  id={props.selectedAnswer.id}
-                  material={props.selectedAnswer.material}
-                  wordClassName="c-word c-word--blank"
-                />
-              </div>
-            } else {
-              return <FillTheBlankWordDropZone
-                  className="c-blank-drop-zone"
-                  key={index}
-                  dropItem={(answerId) => { props.linkWord(answerId) }}
-                >
-                  <div className="c-blank" />
-                </FillTheBlankWordDropZone>
-            }
-          } else
-
-          return <Word className="c-word c-word--fill" key={index} material={wordHtml} />
-        })
+export default class FillTheBlankWordChain extends React.Component {
+  static propTypes = {
+    linkWord: React.PropTypes.func.isRequired,
+    sentenceWords: React.PropTypes.array.isRequired,
+    selectedAnswer: React.PropTypes.object
+  }
+
+  constructor() {
+    super();
+
+    this.state = {
+      wrapIndexes: [0]
+    }
+
+    this.wordRefs = {};
+  }
+
+  componentDidMount() {
+    // We are doing this because for some reason the words have not finished
+    // rendering when componentDidMount is called. A 1ms set timeout is enough time.
```

Comment:
Add a comment indicating that we do this because we need to calculate the widths.